includes/admin/class-admin.php

```diff
@@ -1144,11 +1157,11 @@ public function check_setup() {
             if ( wpbdp_get_option( 'require-login' )
                 && ! get_option( 'users_can_register' )
                 && ! get_user_meta( get_current_user_id(), 'wpbdp_notice_dismissed[registration_disabled]', true ) ) {
-                    $this->messages[] = array(
-                        str_replace( array( '[', ']' ), array( '<a href="' . esc_url( admin_url( 'options-general.php' ) ) . '">', '</a>' ), _x( 'We noticed you want your Business Directory users to register before posting listings, but Registration for your site is currently disabled. Go [here] and check "Anyone can register".', 'admin', 'business-directory-plugin' ) ),
-                        'error dismissible',
-                        array( 'dismissible-id' => 'registration_disabled' ),
-                    );
+					$this->messages[] = array(
+						sprintf( _x( 'We noticed you want your Business Directory users to register before posting listings, but Registration for your site is currently disabled. Go %1$shere%2$s and check "Anyone can register".', 'admin', 'business-directory-plugin' ), '<a href="' . esc_url( admin_url( 'options-general.php' ) ) . '">', '</a>' ),
```

Comment:
For PHPCS, sprintf( should be on its own line for better readability. This is really long to keep all on one line.